packages/core/admin/admin/src/content-manager/components/BlocksEditor/plugins/withDataInsertion.js

```diff
@@ -0,0 +1,31 @@
+import { insertLink } from '../utils/links';
+
+/**
+ * This plugin is used to add data as a clickable link if its a valid URL
+ *
+ * @param {import('slate').Editor} editor
+ */
+
+const withDataInsertion = (editor) => {
+  const { insertData } = editor;
+
+  editor.insertData = (data) => {
+    const pastedText = data.getData('text/plain');
+
+    if (pastedText) {
+      const isValidUrl = /^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$/i.test(pastedText);
```

Comment:
looks like this doesn't match HTTP (only HTTPS). I'm also not sure we need a complex regex here instead of just startsWith. Could you share where it comes from?

Reply:
Couldn't you also just try to parse it with the URL web API? 🤔

Reply:
yes that sounds good. Also there are other protocols than http(s) and ftp, and we have no reason to refuse them in my opinion. For example mailto as seen in #18562. I just checked and using the URL web api allows it

Reply:
@joshuaellis Thank you for the suggestion, I am validating URL with try catch, as URL.canParse() is not supported by many browsers yet, https://developer.mozilla.org/en-US/docs/Web/API/URL/canParse_static